feat(@desktop/wallet): Implement loading animation

Khushboo-dev-cpp opened the pull request:
### What does the PR do

Implements generic loading state support for:
1. StatusBaseText
2. StatusLIstItem
3. StatusSmartIdenticon
as per new designs https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=9476%3A104890&t=PI0EIQ2jwLFNhOxf-4

### Affected areas
1. StatusBaseText
2. StatusLIstItem
3. StatusSmartIdenticon

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://user-images.githubusercontent.com/60327365/210858286-ef82bd4b-30bc-411c-bed0-5eeaf4c23579.mov

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9093b8f9 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/imports/job/PR-8964/1/) | 2023-01-04 17:28:40 | ~5 min | `imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/imports/job/PR-8964/1//consoleText) |

Khushboo-dev-cpp commented:
@stefandunca @dlipicar @alaibe this is still a draft and I still need to clean up the code, but have a look at the video and the code. After taking into consideration into points from you guys, I think this is the best way forward. 

1. Because we dont want to maintain two pieces of code that are meant to look similar but dont because say the the StatusLIstItem design changes. We will need to update any design changes in both the "actual" and "loading" version of the item. 
2. Also considering that this loading state is meant to be generic all throughout our app, this approach seems beneficial to us.

Let me know your thoughts and we can discuss it in tomorrow stand up as well

Khushboo-dev-cpp commented:
@alexandraB99 ! we actually need help from the UI team to let us know if this approach is something we agree with. Please check the design links I have shared in the ticket and the video attached.

Even though this is only currently showing up for wallet, it was discussed that will be the approach to loading states in application eventually which is why I think embedding it with the basic components is a good idea. Let me know :)

alexandraB99 commented on ui/StatusQ/src/StatusQ/Components/LoadingComponent.qml:
```diff
@@ -0,0 +1,39 @@
+
+Rectangle {
+    id: root
+    property bool loading
```
could you please make this a `Controls` with `background` and `contentItem` properties set instead?

alexandraB99 commented on ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:
```diff
@@ -0,0 +1,29 @@
+            loading: root.loading
+        }
+    }
+    TextMetrics {
```
is this used somewhere?

Khushboo-dev-cpp replied:
yes I need it for getting the height of the letter

alexandraB99 commented on ui/StatusQ/src/StatusQ/Core/Theme/StatusDarkTheme.qml:
```diff
@@ -62,6 +62,9 @@ ThemePalette {
 
+//    statusLoadingHighlight: getColor('white', 0.07)
+//    statusLoadingHighlight2: getColor('lightPattensBlue', 0.03)
+
```
please remove if not needed/used

caybro commented on ui/StatusQ/src/StatusQ/Controls/StatusTextWithLoadingState.qml:
```diff
@@ -0,0 +1,29 @@
+    property color customColor: Theme.palette.directColor1
+    color: loading ? "transparent" : customColor
+    Loader {
+        anchors.left: parent.left
```
It's better to set both anchors here

Khushboo-dev-cpp replied:
wdym?

caybro replied:
Also `anchors.right`

Khushboo-dev-cpp replied:
wont be needed as I specified a width in the loaders source component

caybro replied:
OK

caybro reviewed (commented):
Overall nice work!

Please make it a proper `Control` as Alex mentioned; bonus points for putting this into Storybook ;)

alexandraB99 commented:
> Overall nice work!
> 
> Please make it a proper `Control` as Alex mentioned; bonus points for putting this into Storybook ;)

@caybro good point, should we just not add it to sandbox and just add it in storybook instead? wdyt?

Khushboo-dev-cpp commented:
@alexandraB99 @caybro addresses all the comments for the bonus points for storybook :D

let me know if we can proceed with this task, also I am guessing we will be redoing the StatusQ components after MVP and new design system maybe we can move all of the demos from sandbox to story book then?

Write now its all here and there, I am also adding the loading state to StatusLIstItem and for that I need to use sandbox as I dont thats migrated to storybook yet?
